crates/sui-core/src/transaction_manager.rs

```diff
@@ -138,12 +145,24 @@ impl TransactionManager {
                 continue;
             }
 
-            let mut package_inputs = cert.data().intent_message.value.type_argument_packages();
-            package_inputs.extend_from_slice(&cert.data().intent_message.value.input_objects()?);
+            let mut inputs = cert.data().intent_message.value.input_objects()?;
+
+            // if effects indicate a success then we need to add and wait for argument packages,
+            // otherwise we can skip
```

Comment:
Is it possible to have cases where having input types available results in one error (and more validators got lucky), and without input types the error becomes `EntryTypeArgumentError` (where a unlucky fullnode can hit)? Maybe we can be more exact here to match EntryTypeArgumentError { argument_idx, ModuleNotFound} and pass the input type ID to txn mgr, but I'm not 100% sure if that is either bulletproof or necessary.

Reply:
But I guess it is possible to have multiple args missing input types, so we do need to avoid adding any input type dependency if there is a ModuleNotFound error.